### PR TITLE
Fix quarantined tests - TestCliWebServer (v1-10-test)

### PR DESCRIPTION
**Pull request by mik-laj**
I'm starting work on preparing a patch for the master branch as well.

It turned out that the test for webserver in deamon mode was not flaky, but there was a real bug there, but it was never printed on the screen.

---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Target Github ISSUE in description if exists
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Though some tests are still failing

**Comment by mik-laj** [comment]
@potiuk  I fixed it now.

**Review by potiuk — APPROVED** [review]

**Comment by kaxil** [comment]
Tests are still failing

**Comment by mik-laj** [comment]
@kaxil I know :_;

**Comment by kaxil** [comment]
> @kaxil I know :_;

:D Although all the tests in v1-10-test passed: https://github.com/apache/airflow/commits/v1-10-test (https://github.com/apache/airflow/runs/824432425)

**Comment by mik-laj** [comment]
@kaxil We have Python 3.8 compatibility issues: https://apache-airflow.slack.com/archives/CCPRP7943/p1593558215208800

**Comment by ashb** [comment]
212 commits -- something has gone wrong with this PR (so we can't review it)

**Comment by kaxil** [comment]
Sorry, it was because I force-pushed v1-10-test to remove all fixups as we are almost ready

Can you please rebase on latest v1-10-test

**Comment by mik-laj** [comment]
@kaxil I made the necessary changes to make the tests stable and did rebase.

**Comment by kaxil** [comment]
> @kaxil I made the necessary changes to make the tests stable and did rebase.

Thanks, I will merge as soon as the CI is complete

**Comment by mik-laj** [comment]
I forgot to generate requirements for Python 3.5. I'm working on it.

File: requirements/requirements-python3.8.txt [hunk 1]
```diff
@@ -50,7 +50,7 @@ aws-xray-sdk==2.6.0
 azure-common==1.1.25
 azure-cosmos==3.1.2
 azure-datalake-store==0.0.48
-azure-mgmt-containerinstance==1.5.0
+azure-mgmt-containerinstance==2.0.0
```
Comment (mik-laj) [inline requirements/requirements-python3.8.txt]: It broke my build. I'm working on a fix.

Comment (mik-laj) [reply requirements/requirements-python3.8.txt]: I just pushed a fix.

**Pull request merged**